storage/src/storage.rs

```diff
@@ -1,25 +1,23 @@
 //! Module containing a Storage generic trait that can be implemented for different specific storage
 //! backends.
 
-use crate::error::Result;
+use crate::error::StorageResult;
 
 /// This is a generic trait that exposes a very simple key/value CRUD API for data storage.
 /// This trait can be easily implemented for any specific storage backend solution (databases,
 /// volatile memory, flat files, etc.)
-pub trait Storage<ConnData, Key, Value> {
-
+pub trait Storage<Key, Value> {
     /// Storage object constructor.
     /// `connection_data` can be used for passing credentials, urls, paths, etc. down to the storage
     /// backend.
-    fn new(connection_data: ConnData) -> Result<Box<Self>>;
+    fn new(connection_data: String) -> StorageResult<Box<Self>>;
```

Comment:
As said before, we want `Storage` to be generic over different storage backends that may require different connection data types. Forcing a `String` here is doing the opposite.